nixos/modules/services/web-apps/dolibarr/default.nix

```diff
@@ -0,0 +1,237 @@
+              '';
+            in (if (cfg.initialDbPasswordFile != null) then ''
+              tmp_script_file=$(mktemp)
+              echo ${escapeShellArg initialScript} > $tmp_script_file
+              ${pkgs.replace-secret}/bin/replace-secret '${passwordPlaceholder}' '${cfg.initialDbPasswordFile}' $tmp_script_file
+              cat $tmp_script_file | ${config.services.mysql.package}/bin/mysql -u root -N
+              rm -f $tmp_script_file
+            '' else "");
+        in ''
+          #!${pkgs.runtimeShell}
+          # Setup database password
+          echo 'SELECT User FROM mysql.user WHERE User = "${app}";' | ${config.services.mysql.package}/bin/mysql -u root -N | grep '${app}'
+          if [ ! $? = 0 ]; then
+            ${mysqlPasswordActivation}
+          fi
+        '';
+        startScript = pkgs.writeScriptBin "${app}-mysql-db-init-start" script;
+      in {
+        Type = "oneshot";
+        ExecStart = "${startScript}/bin/${app}-mysql-db-init-start";
+      };
+    };
+
+    systemd.services.nginx = {
+      wants = [ "${app}-mysql-db-init.service" ];
+      after = [ "${app}-mysql-db-init.service" ];
+    };
+    systemd.services.phpfpm-dolibarr = {
+      serviceConfig.ReadWriteDirectories = [
+        "/etc/${app}"
+        "/var/lib/${app}/documents"
+      ];
+    };
+
+    system.activationScripts.dolibarr-init-script = let
+      initConfig = ''
+        <?php
+
+        $dolibarr_main_url_root='http://${cfg.domain}';
+        $dolibarr_main_document_root='${webroot}';
+        $dolibarr_main_url_root_alt='/custom';
+        $dolibarr_main_document_root_alt='${webroot}/custom';
+        $dolibarr_main_data_root='/var/lib/${app}/documents';
+        $dolibarr_main_db_host='localhost';
+        $dolibarr_main_db_port='3306';
+        $dolibarr_main_db_name='${app}';
+        $dolibarr_main_db_prefix='llx_';
+        $dolibarr_main_db_user='${app}';
+        $dolibarr_main_db_pass='${passwordPlaceholder}';
+        $dolibarr_main_db_type='mysqli';
+        $dolibarr_main_db_character_set='utf8';
+        $dolibarr_main_db_collation='utf8_unicode_ci';
+        $dolibarr_main_authentication='${app}';
+        $dolibarr_main_prod='0';
+        $dolibarr_main_force_https='0';
+        $dolibarr_main_restrict_os_commands='mysqldump, mysql, pg_dump, pgrestore';
+        $dolibarr_nocsrfcheck='0';
+        $dolibarr_main_instance_unique_id='${uniqueIdPlaceholder}';
+        $dolibarr_mailing_limit_sendbyweb='0';
+        $dolibarr_main_distrib='standard';
+      '';
+    in stringAfter [ "etc" "groups" "users" ] ''
+      # Setup folders in /var/lib/${app}/documents
+      mkdir -p /var/lib/${app}/documents/{mycompany,medias,users,facture,propale,ficheinter,produit,doctemplates}
+      chown -R ${app}:${app} /var/lib/${app}
+      chmod -R 0700 /var/lib/${app}
+
+      # Setup config file
+      tmp_hash=$(mktemp)
+      head -n 100 /dev/random| md5sum | cut -d' ' -f1 > $tmp_hash
```

Comment:
Do we need to handle this id securely? Does it need to be an md5 hash? Otherwise we could simplify this be using uuidgen.